pkg/bootstrapper/bootstrapper.go

```diff
@@ -19,6 +19,7 @@ type ClusterClient interface {
 	CreateBootstrapCluster(ctx context.Context, clusterSpec *cluster.Spec, opts ...BootstrapClusterClientOption) (kubeconfig string, err error)
 	DeleteBootstrapCluster(ctx context.Context, cluster *types.Cluster) error
 	WithExtraDockerMounts() BootstrapClusterClientOption
+	WithExtraPortMappings([]int) BootstrapClusterClientOption
```

Comment:
I don't think you have much of a choice with respect to doing it this way but I thought I'd highlight, nonetheless, that there's a small code smell here.

We have a `Bootstrapper` that is arguably a premature abstraction which adds a level of indirection to a Kind implementation of a bootstrapper. The introduction of extra port mappings, and possibly other options here too, are seemingly indicative of an abstraction tightly coupled to an implementation.

Reply:
Yeah I agree. We can create a backlog item to refactor this since this is mainly out of scope for this PR